### PR TITLE
Add mamma: streaming multiview SMPL-X body capture (MAMMA port)

### PR DESCRIPTION
**Pull request by pablo-rerun**
## What

Adds `packages/mamma` — a causal **streaming** port of the MAMMA multiview body-capture pipeline: NVDEC decode → GPU resize → SAM2 tracking → MammaNet dense 2D landmarks → GPU triangulation → sliding-window SMPL-X fit → Rerun, in one resident process with **no disk writes anywhere in the loop** (vs the original 5-subprocess offline DAG that round-tripped NPZ/PNG between stages).

Handles **single- and multi-person** capture with an auto-detected subject count. Also in this PR:

- vendors the cjaverliat/sam2 fork (streaming predictor + forgetful-memory bank) as `packages/sam2-streaming`;
- extends **simplecv** with decode-time resize (`resize_hw` on the TorchCodec readers) and an H.264 `VideoCodecChoice`;
- adds a **calibration-free front end** for uncalibrated synced video (VGGT + MoGe-v2 monoprior pre-pass, then confidence-weighted human-keypoint bundle adjustment);
- adds **iPhone-dataset tooling**: a quality sweep that gates each scene's PVE (p95 & p99 ≤ 30 mm) and per-camera realtime against the dataset's `pred` reference, then serves all scenes as a Rerun **catalog** (base + props + default blueprint, sortable by subject count / error).

## Verification — `pixi run mamma-goal-check` exits 0 (6/6 PASS)

| Clause | Criterion |
|---|---|
| golden | MPJPE/PVE vs the frozen original output, crossing_arms frames 60:90, ≤ 30 mm |
| dynamic | running_jumping per-frame PVE p95 AND max ≤ 30 mm vs the original DAG |
| realtime | both clips: wall ≤ 2× clip duration (≥ 50% of realtime) incl. Rerun logging |
| datasets | HOCap + Assembly101 ingest end-to-end → validated RRDs |
| no-writes | streaming loop creates no files (runtime check + static write-call audit) |
| hygiene | ruff + pyrefly + pytest clean in the `mamma-dev` env |

iPhone sweep: **29/42 scenes** pass PVE p95 & p99 ≤ 30 mm vs `pred` (every single-person scene + several multi-person); the remaining failures are hard multi-person heavy-occlusion cases.

## Performance

The original 5-subprocess offline DAG ran at **~12 s/frame** (~1h47m for the 478-frame running_jumping clip — hundreds of × off realtime). This pipeline runs at **~40.7 ms/frame** (14.79 s wall for the 363-frame crossing_arms clip, Rerun logging included) — every optimization re-verified against the golden gate: CUDA-graphed sliding-window fitter (graphed 300-iter bootstrap), 512-vertex sampled SMPL-X with a sync-free rigid chain, TensorRT MammaNet engine (3.88 ms vs 15.8 eager), SAM2-fork B=n_cams batched propagation, multiprocess NVDEC decode (~400 cam-fps around torchcodec 0.10's single-decoder cache), async NVENC H.264 → `rr.VideoStream` logging.

Defaults favor quality: `track_stride=1` keeps masks pixel-identical to per-frame tracking; the speed config is one flag away (`--tracker.track-stride 4`, 11.76 s, with transient mask collapse on fast motion). The `mp_decode` worker is a torchcodec-0.10 workaround — `docs/cuda13-migration.md` is the validated recipe to remove it (torch 2.12 + torchcodec 0.13 on CUDA 13, aarch64-capable for DGX Spark).

## Data & licensing

MAMMA inputs, MammaNet weights, and SMPL-X are license-gated: everything lives in the **private** HF dataset repo `pablovela5620/mamma-streaming-data`, fetched by the `_download-mamma-*` pixi tasks. Nothing gated is committed; the original repo was used read-only as the golden-artifact generator.

## Docs

`packages/mamma/implementation-notes.html` — running decision log (deviations, perf rounds, rerun-viewer-validation evidence); served out-of-band (gitignored — large embedded media). `docs/cuda13-migration.md` — the deferred CUDA-13 / torch-2.12 / torchcodec-0.13 migration plan.

### CODE REVIEW

**Comment by pablovela5620** [comment]
Post-PR adversarial review (7 finder angles + per-candidate verification) landed in a32c3f6:

**Fixed**
- `StreamingPipeline.run()` is now exception-safe: the fit worker (non-daemon thread) is always shut down, the decode prefetcher is released via generator `finally`, and decode workers are closed on the error path — a mid-loop exception can no longer hang interpreter exit or leak NVDEC/CUDA contexts.
- `anchor3d_loss` gets the same `nan_to_num` guard as `reprojection_loss` (NaN × 0-mask = NaN; a degenerate-but-valid triangulation could poison the fit).
- benchmark warmup: close + recording restore in `finally`; per-tick numpy import hoisted; `emit_stride` docstring now warns about stale `frame_idx` on skipped ticks.

**Checked and refuted** (for reviewer peace of mind): StreamLogger cross-thread `set_time` race (rerun 0.33 clocks are thread-local), silent frame drops on length-mismatched cameras (frame_count is min-clamped at reader init), TRT batch >4 (guard routes to eager correctly).

Golden gate unchanged after fixes: 23.3mm MPJPE / 21.4mm PVE. Benchmark fix-cost A/B-verified as zero (pre- and post-fix identical under load; canonical 14.79s sweep stands).

File: packages/sam2-streaming/sam2/modeling/sam2_prompt.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        if points_coords is not None and points_labels is None:
+            raise ValueError(
+                "points_labels must be provided if points_coords is provided"
+            )
+        
+        if points_coords is not None and (points_coords.ndim != 2 or points_coords.shape[1] != 2):
+            raise ValueError(f"Expected points_coords to be of shape (N, 2), got {points_coords.shape}")
+        
+        if points_labels is not None and (points_labels.ndim != 1 or points_labels.shape[0] != points_coords.shape[0]):
+            raise ValueError(f"Expected points_labels to be of shape (N,), got {points_labels.shape}")
```
[the remark this answers is not in the document]
Comment (pablo-rerun) [reply packages/sam2-streaming/sam2/modeling/sam2_prompt.py:35]: Skipped (intentionally). Real latent bug, but `sam2_prompt.py` is vendored fork code and this is the **points-prompt** path — `mamma` drives SAM2 with **box prompts only**, so it never executes. Fixing it would add divergence from the vendored upstream for no runtime effect here. Confirmed with an independent second review.

File: packages/sam2-streaming/sam2/build_sam.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+from omegaconf import OmegaConf
+
+import sam2
+from sam2.sam2_generic_video_predictor import SAM2GenericVideoPredictor
+
+# Check if the user is running Python from the parent directory of the sam2 repo
+# (i.e. the directory where this repo is cloned into) -- this is not supported since
+# it could shadow the sam2 package and cause issues.
+if os.path.isdir(os.path.join(sam2.__path__[0], "sam2")):
+    # If the user has "sam2/sam2" in their path, they are likey importing the repo itself
```
[the remark this answers is not in the document]
Comment (pablo-rerun) [reply packages/sam2-streaming/sam2/build_sam.py:22]: Skipped. This comment block is **stock upstream Facebook sam2** text; keeping it verbatim avoids gratuitous divergence from the vendored source (this is a slimmed vendored fork).

File: packages/sam2-streaming/sam2/build_sam.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    if ckpt_path is not None:
+        sd = torch.load(ckpt_path, map_location="cpu", weights_only=True)["model"]
+        missing_keys, unexpected_keys = model.load_state_dict(sd)
+        if missing_keys:
+            logging.error(missing_keys)
+            raise RuntimeError()
+        if unexpected_keys:
+            logging.error(unexpected_keys)
+            raise RuntimeError()
+        logging.info("Loaded checkpoint sucessfully")
```
[the remark this answers is not in the document]
Comment (pablo-rerun) [reply packages/sam2-streaming/sam2/build_sam.py]: Fixed in 00038fe. `_load_checkpoint` now raises `RuntimeError` with the missing/unexpected key lists, and 'sucessfully' → 'successfully'.

File: packages/mamma/tools/register_catalog.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        gate: dict = json.loads(gate_path.read_text()) if gate_path.exists() else {}
+        if gate.get("error"):  # crashed/incomplete scene (sweep wrote a synthetic error gate) — don't register garbage
+            print(f"  SKIP {name}: {gate['error']}")
+            continue
+        n_people: int = int(gate.get("n_people") or len(list((config.data_root / cat_ / scene / "pred").glob("params_*.npz"))) or 1)
+
+        rid: str = ds.register([f"file://{p}"], layer_name="base", on_duplicate=replace).wait().segment_ids[0]
+
+        rr.init(config.application_id, recording_id=rid)
+        rr.send_recording_name(name)
```
[the remark this answers is not in the document]
Comment (pablo-rerun) [reply packages/mamma/tools/register_catalog.py]: Fixed in 00038fe. Added `strict=True` to `rr.init`. The catalog props logging is simple, well-typed scalars/strings (all None-guarded), so strict is safe and matches the RerunTyroConfig convention other mamma CLIs use.

File: packages/mamma/src/mamma/engine/mp_decode.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+                raise
+            if kind == "error":
+                self.close()
+                raise RuntimeError(f"decode worker failed to initialize ({payload}):\n{tb}")
+            assert kind == "ready", f"unexpected startup message from decode worker: {kind!r}"
+
+    def iter_chunks(self, start_frame: int, frame_count: int):
+        """Yield ``(chunk_start, [per-camera UInt8 CUDA tensors])`` in order."""
+        for cmd_q in self._cmd_queues:
+            cmd_q.put((start_frame, frame_count, self.chunk_size))
```
[the remark this answers is not in the document]
Comment (pablo-rerun) [reply packages/mamma/src/mamma/engine/mp_decode.py]: Fixed in 00038fe. Renamed the `frame_count` param (and the worker job tuple) to `frame_end` and documented it as an exclusive end index. No active bug today — `pipeline.py` already passes an end index — but the name was a real hazard, so the rename stands.

File: packages/mamma/src/mamma/engine/presets.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            redetect_interval=45,
+            reprompt_alive_on_routine=True,
+            transient_hold=True,
+            track_stride=1,
+        ),
+        fitter=FitterConfig(tick_iters=24, emit_stride=2, lbfgs_polish=False),
+        resize_hw=(720, 1280),
+        hires_crops=False,
+        realtime_fraction=0.75,
+    )
```
[the remark this answers is not in the document]
Comment (pablo-rerun) [reply packages/mamma/src/mamma/engine/presets.py:112]: Fixed in 00038fe. Lowered `fast_preset` `realtime_fraction` 0.75 → 0.40 to match the documented ~40% operating point so `--preset fast` no longer fails its own gate, and reworded the docstring to drop the aspirational-75% contradiction.
